libpages/server.go

```diff
@@ -332,8 +369,18 @@ func ListenAndServe(ctx context.Context,
 		shutdownCtx, cancel := context.WithTimeout(
 			context.Background(), gracefulShutdownTimeout)
 		defer cancel()
-		httpServer.Shutdown(shutdownCtx)
+		httpsServer.Shutdown(shutdownCtx)
+		httpRedirectServer.Shutdown(shutdownCtx)
```

Comment:
Why aren't these the same server? Seems wasteful to have 2 sets of listeners for it.

Reply:
> Why aren't these the same server? 

They are not same server because the two servers handle requests in different ways. I don't see how different handlers on same path can be put into same `http.Server`. Or did you mean something different?

> Seems wasteful to have 2 sets of listeners for it.

We need a different listener for each port we are listening on right?

Reply:
Argh brain fart, I forgot that they need to be on separate ports.

Reply:
Must have been too much holiday food :D

Reply:
Also: I'm used to having a server of some kind proxying requests, so the application port is usually arbitrary.